bundles/statistics/statsgrid/view/Diagram/DiagramFlyout.jsx

```diff
@@ -52,15 +55,23 @@ const DiagramFlyout = ({ state, controller }) => {
                     options={state.indicators?.map(indicator => ({ value: indicator.hash, label: indicator.labels?.full }))}
                     onChange={(value) => controller.setActiveIndicator(value)}
                     value={state.activeIndicator}
+                    placeholder={<Message messageKey='panels.newSearch.selectIndicatorPlaceholder' />}
                 />
                 <StyledSelect
                     filterOption={false}
                     options={sortOptions}
                     onChange={(value) => controller.setSortOrder(value)}
                     value={state.sortOrder}
+                    placeholder={<Message messageKey='datacharts.sorting.desc' />}
                 />
            </Selections>
-           <Diagram chartData={state.chartData} sortOrder={state.sortOrder} />
+           {!state.chartData?.data ? (
+                <NoDataMessage>
+                    <Message messageKey='datacharts.nodata' />
+                </NoDataMessage>
+           ) : (
+                <Diagram chartData={state.chartData} sortOrder={state.sortOrder} />
+           )}
```

Comment:
Should we have this on Diagram-component instead so it can manage by itself even if it doesn't receive any data and doesn't rely on other components handling this?